Source/Core/Resource.js

```diff
@@ -1775,6 +1860,47 @@ define([
         }
 
         image.src = url;
+    }
+
+    Resource._Implementations.createImage = function(url, crossOrigin, deferred, flipY) {
+        if (!FeatureDetection.supportsCreateImageBitmap()) {
+            loadImageElement(url, crossOrigin, deferred);
+            return;
+        }
+
+        // Passing an Image to createImageBitmap will force it to run on the main thread
+        // since DOM elements don't exist on workers. We convert it to a blob so it's non-blocking.
+        // See:
+        //    https://bugzilla.mozilla.org/show_bug.cgi?id=1044102#c38
+        //    https://bugs.chromium.org/p/chromium/issues/detail?id=580202#c10
+        Resource.supportsImageBitmapOptions()
+            .then(function(result) {
+                // We can only use ImageBitmap if we can flip on decode.
+                // See: https://github.com/AnalyticalGraphicsInc/cesium/pull/7579#issuecomment-466146898
+                if (!result) {
+                    loadImageElement(url, crossOrigin, deferred);
+                    return;
+                }
+
+                Resource.fetchBlob({
```

Comment:
This should have a return statement and then you should move all of the following `then` calls up a level.

Reply:
Can you explain why? I'm assuming this is what you mean:

```
Resource.supportsImageBitmapOptions()
    .then(function(result) {
        if (!result) {
            loadImageElement(url, crossOrigin, deferred);
            return;
        }

        return Resource.fetchBlob({
            url: url
        });
    })
    .then(function(blob) {
        return Resource._Implementations.createImageBitmapFromBlob(blob, flipY);
    })
    .then(deferred.resolve).otherwise(deferred.reject);
```

This will run the `.then(function(blob)` anyway even if we `loadImageElement` and don't fetch the blob, and then `blob` will be `undefined`. 

I can just move the `.then(deferred.resolve).otherwise(deferred.reject);` outside and I that should work fine.

Reply:
I can see `Resource.prototype.fetchImage` checks if the result is defined at each step in the promise chain to make sure it doesn't run when it terminates early like that. I can follow that pattern.

Reply:
Because if for some reason `supportsImageBitmapOptions` rejected, you would  never call `deferred.reject` (or `deferred.resolve`).

Reply:
> I can see `Resource.prototype.fetchImage` checks if the result is defined at each step in the promise chain to make sure it doesn't run when it terminates early like that. I can follow that pattern.

I'm not sure what you mean by this, please stop by if you still have questions.

Reply:
Thanks for the explanation. I'm pretty sure I've got it locked down now. What I meant was, Resource has a similar promise chain when fetching an image. When we return undefined, all the subsequent then's are still going to run. So you need to add a check at each subsequent then() that the result is defined. 

Here is where Resource is doing this: 

https://github.com/AnalyticalGraphicsInc/cesium/blob/8bc364ce5eca3c42d84fe21386b0d5a50be82c2d/Source/Core/Resource.js#L886-L889

I just followed a similar pattern in `createImage`:

https://github.com/AnalyticalGraphicsInc/cesium/blob/a6324e2bcf8b3547b4adb589e7ee789321997678/Source/Core/Resource.js#L1855-L1882

Notice that `deferred.resolve` is only called in the ImageBitmap codepath. Since `loadImageElement` calls it on the `Image` on `onload`.

Reply:
I think this whole file could probably use some clean-up.  Some of the (old) if checks make reading the code pretty confusing.  No action on your part, and this is nothing you did; just mentioning it because I may open a small clean-up PR to take a pass (after merging this).